internal/contour/cluster_test.go

```diff
@@ -240,6 +242,383 @@ func TestClusterVisit(t *testing.T) {
 				},
 			),
 		},
+		"ingressroute with simple path healthcheck": {
+			objs: []interface{}{
+				&ingressroutev1.IngressRoute{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "simple",
+						Namespace: "default",
+					},
+					Spec: ingressroutev1.IngressRouteSpec{
+						VirtualHost: &ingressroutev1.VirtualHost{
+							Fqdn: "www.example.com",
+						},
+						Routes: []ingressroutev1.Route{{
+							Match: "/",
+							Services: []ingressroutev1.Service{{
+								Name: "backend",
+								Port: 80,
+								HealthCheck: &ingressroutev1.HealthCheck{
+									Path: "/healthy",
+								},
+							}},
+						}},
+					},
+				},
+				service("default", "backend", v1.ServicePort{
+					Name:       "http",
+					Protocol:   "TCP",
+					Port:       80,
+					TargetPort: intstr.FromInt(6502),
+				}),
+			},
+			want: clustermap(
+				&v2.Cluster{
+					Name: "default/backend/80",
+					Type: v2.Cluster_EDS,
+					EdsClusterConfig: &v2.Cluster_EdsClusterConfig{
+						EdsConfig:   apiconfigsource("contour"), // hard coded by initconfig
+						ServiceName: "default/backend/http",
+					},
+					ConnectTimeout: 250 * time.Millisecond,
+					LbPolicy:       v2.Cluster_ROUND_ROBIN,
+					HealthChecks: []*envoy_api_v2_core4.HealthCheck{{
+						Timeout: &google_protobuf.Duration{
```

Comment:
Hmm, something is weird here with the grpc types, look at ConnectTimeout: that takes a time.Duration, why do these fields take a protobuf.duration? is it because they are optional? (i've seen other stuff in envoy where a pointer to zero means infinity but a nil pointer means -- effectively -- zero.